libp2p/network/swarm.py

```diff
@@ -71,27 +255,47 @@ class Swarm(Service, INetworkService):
     peerstore: IPeerStore
     upgrader: TransportUpgrader
     transport: ITransport
-    # TODO: Connection and `peer_id` are 1-1 mapping in our implementation,
-    #   whereas in Go one `peer_id` may point to multiple connections.
-    connections: dict[ID, INetConn]
+    # Enhanced: Support for multiple connections per peer
+    connections: dict[ID, INetConn]  # Backward compatibility
```

Comment:
Don't know if keeping 1-1 mapping at this level is the right architectural ways ( we have to compare pro/cons, see go and javascript implementations )